sklearn/datasets/_base.py

```diff
@@ -340,7 +340,8 @@ def load_csv_data(
         Description of the dataset (the content of `descr_file_name`).
         Only returned if `descr_file_name` is not None.
     """
-    with _open_text(data_module, data_file_name) as csv_file:
+    data_path = resources.files(data_module) / data_file_name
+    with data_path.open("r") as csv_file:
```

Comment:
I would rather never use open in text mode without specifying the expected encoding:

The `pathlib.Path.open` doc refers to the `open` built-in whose docstring states:

> In text mode, if encoding is not specified the encoding used is platform
dependent: locale.getencoding() is called to get the current locale encoding.

Having a dataset loading behavior that depends on the OS configuration is the user is problematic because our datasets are all encoding with a fixed encoding.

I would be in favor of adding `encoding="utf-8"` as additional kwarg to `load_csv_data`, pass that to the underlying call and make it possible to pass an alternative encoding for specific datasets that use `load_csv_data` on a case by case basis.